nncf/quantization/algorithms/smooth_quant/algorithm.py

```diff
@@ -89,22 +88,24 @@ def _set_backend_entity(self, model: TModel) -> None:
                 "Cannot return backend-specific entity because {} is not supported!".format(model_backend)
             )
 
-    def _apply(
+    def apply(
         self,
         model: TModel,
+        graph: NNCFGraph,
         statistic_points: Optional[StatisticPointsContainer] = None,
         dataset: Optional[Dataset] = None,
     ) -> TModel:
         if self._alpha < 0:
             nncf_logger.info("Skipping SmoothQuant algorithm because alfa parameter is negative.")
             return model
 
-        nncf_graph = NNCFGraphFactory.create(model)
-        nodes_to_smooth_data = self._get_nodes_to_smooth_data(nncf_graph)
+        self._set_backend_entity(model)
```

Comment:
`apply(...)` method can be called before `get_statistics(...)` method which set backend entity. I know, you will say it cann't be but we don't have mechanism to avoid it. In this case, the user probably got error that backend entity is None and user will be confused because algorithm does not have methods to set backend entity. If we set backend entity into `apply` method the user got error relatives to statistics or something else... I think it will be more clear.